src/DynamoPackages/PackageLoader.cs

```diff
@@ -260,11 +262,38 @@ private void TryLoadPackageIntoLibrary(Package package)
                         }
                         catch (LibraryLoadFailedException ex)
                         {
+                            // Managed exception
+                            // We can still try to load other parts of the package
                             Log(ex.GetType() + ": " + ex.Message);
                         }
+                        catch (DynamoServices.AssemblyBlockedException)
+                        {
+                            blockedAssemblies.Add(assem.Assembly);
+                        }
+                        catch (Exception)
+                        {
+                            failedNodeLibs.Add(assem.Assembly);
+                        }
                     }
                 }
 
+                if (loadedNodeLibs.Count > 0)
+                {
+                    // Try to load any valid node libraries regardless package state
+                    PackagesLoaded?.Invoke(loadedNodeLibs);
+                }
+
+                if (blockedAssemblies.Count > 0)
+                {
+                    throw new Exception("The following assemblies are blocked : " + string.Join(", ", blockedAssemblies.Select(x => Path.GetFileName(x.Location))));
+                }
+
+                // Generic fatal error
+                if (failedNodeLibs.Count > 0)
+                {
+                    throw new Exception("Failed to load the following assemblies : " + string.Join(", ", failedNodeLibs.Select(x => Path.GetFileName(x.Location))));
```

Comment:
Shouldn't this be `new LibraryLoadFailedException`?

Reply:
well..it could be..but I don't think it matters...since we do not catch that in any special way in this try catch